libgalois/include/galois/OpLog.h

```diff
@@ -19,6 +19,19 @@ enum class OpTypes {
   kOpEdgePropVal,
 };
 
+const char* const optypes_enum2str[] = {
+    "Invalid",     /* kInvalid */
+    "NodeAdd",     /* kOpNodeAdd */
+    "NodeDel",     /* kOpNodeDel */
+    "EdgeAdd",     /* kOpEdgeAdd */
+    "EdgeDel",     /* kOpEdgeDel */
+    "NodePropDel", /* kOpNodePropDel */
```

Comment:
For code like this I'd recommend this pattern:
```
std::string OpTypeToString(OpType t) {
  switch (t) {
    case OpType::kInvalid:
      return "Invalid";
    ...
  }
}
```

it is more robust to reordering and newer compilers will warn you if you miss a case.

It also wouldn't hurt to move implementation details like the body of `OpTypeToString` to a separate .cpp file. Though that can be done later.

Reply:
I think using string arrays to translate enum-s to their correspondent names is quite common (everywhere in the Linux kernel, at least). People just put the two classes together and manually maintain their consistency. (Maybe it's not common for C++, though.) I can move optypes_enum2str[] to my local code, or use the switch-case instead (I think I'll follow your advice to use switch-case). If so, I'll delete this PR.

Reply:
I think for the current pattern to work, you'd also need to declare `optypes_enum2str[]` `static`; otherwise, we'd define `optypes_enum2str` in each translation unit that included this file and the linker might get angry. We are saved by our current linker policy which explicitly hides symbols not marked `GALOIS_EXPORT` but on macOS that particular feature is not enabled. Currently the macOS builds are failing for a different reason but this is something to be aware of.

> I think using string arrays to translate enum-s to their correspondent names is quite common 

Yeah, it definitely is a reasonable approach, though it is very C-like. C and C++ have many false friends [1]---patterns that make a lot of sense in one language but are out of place in another.

[1] https://en.wikipedia.org/wiki/False_friend

Reply:
I moved all the things to my local header files in the enterprise repository. I used switch-case instead of string array translation. I'll now close this PR. Thanks for all the help!